src/cluster/src/cluster_impl.rs

```diff
@@ -341,10 +346,58 @@ impl Inner {
 
         shards.iter().map(|shard| shard.shard_info()).collect()
     }
+
+    /// Get proper remote compaction node info for compaction offload with meta
+    /// client.
+    async fn get_compaction_node(&self) -> Result<CompactionClientConfig> {
+        let mut config = CompactionClientConfig::default();
+
+        let req = FetchCompactionNodeRequest::default();
+        let resp = self
+            .meta_client
+            .fetch_compaction_node(req)
+            .await
+            .context(MetaClientFailure)?;
+
+        config.compaction_server_addr = resp.endpoint;
+        Ok(config)
+    }
+
+    /// Return a new compaction client.
+    async fn compaction_client(&self) -> CompactionClientRef {
+        // TODO(leslie): impl better error handling with snafu.
+        let config = self
+            .get_compaction_node()
+            .await
+            .expect("fail to get remote compaction node");
+
+        build_compaction_client(config)
+            .await
+            .expect("fail to build compaction client")
+    }
+
+    async fn compact(
```

Comment:
Maybe it is not suitable to add `compact` function to `Cluster`.
I have some thoughts:
- We place all logic into `RemoteCompactionRunner`(pick node, pass task to node)
- Abstract `pick node` logic to a trait `RemoteCompactionNodePicker`, it have extrat benefits for easy testing, too.